fedimint-client/src/lib.rs

```diff
@@ -260,8 +270,16 @@ struct ModuleGlobalClientContext {
 
 #[apply(async_trait_maybe_send!)]
 impl IGlobalClientContext for ModuleGlobalClientContext {
-    fn api(&self) -> &(dyn IFederationApi + 'static) {
-        self.client.api.as_ref()
+    fn api(&self) -> DynFederationApi {
+        self.client.api.clone()
+    }
+
+    fn decoders(&self) -> ModuleDecoderRegistry {
```

Comment:
Probably I should just pass `decoders` to recovery method?

Reply:
@elsirion At some point we've discussed that we should encode inputs/outputs/cis as `Vec<u8>` initially, so code can parse epochs without having to have all decoders. This would be the first place that would benefit from it.

Reply:
@elsirion I can't pass `Decoders` in the recovery state itself, because they are not encodable. And I need it in state transition functions to make progress (decode new epochs).

Are you OK with having it here?